transform_constantfold.ml

```diff
@@ -43,6 +43,9 @@ let const_prop ({formals; instrs} : analysis_input) : instructions option =
     | Assign (y, e) ->
       assert (x <> y);
       Assign (y, replace e)
+    | Array_assign (y, i, e) ->
+      assert (x <> y);
```

Comment:
With the current array semantics (see @o-'s comment https://github.com/reactorlabs/sourir/pull/85#issuecomment-297063689 ) it is illegal to update a `const`-declared array, so we could reject this program at compile-time anyway (we did not think of adding this to `check`, though, and it would become obsolete if we go for a by-reference semantics).